verification/curator-service/ui/src/components/LinelistTable.test.tsx

```diff
@@ -400,3 +400,63 @@ it('can edit a row', async () => {
     const editedRow = await findByText('some edited notes');
     expect(editedRow).toBeInTheDocument();
 });
+
+it('cannot edit data as a reader only', async () => {
+    const cases = [
+        {
+            _id: 'abc123',
+            importedCase: {
+                outcome: 'Recovered',
+            },
+            location: {
+                country: 'France',
+            },
+            events: [
+                {
+                    name: 'confirmed',
+                    dateRange: {
+                        start: new Date().toJSON(),
+                    },
+                },
+            ],
+            notes: 'some notes',
+            sources: [
+                {
+                    url: 'http://foo.bar',
+                },
+            ],
+        },
+    ];
+    const axiosGetResponse = {
+        data: {
+            cases: cases,
+            total: 15,
+        },
+        status: 200,
+        statusText: 'OK',
+        config: {},
+        headers: {},
+    };
+    mockedAxios.get.mockResolvedValueOnce(axiosGetResponse);
+
+    // Load table
+    const { findByText, queryByText } = render(
+        <LinelistTable
+            user={{
+                _id: 'testUser',
+                name: 'Alice Smith',
+                email: 'foo@bar.com',
+                roles: ['reader'],
+            }}
+        />,
+    );
+    expect(mockedAxios.get).toHaveBeenCalledTimes(1);
+    expect(mockedAxios.get).toHaveBeenCalledWith(
+        '/api/cases/?limit=10&page=1&filter=',
+    );
+    const row = await findByText(/abc123/);
+    expect(row).toBeInTheDocument();
+
+    const deleteButton = queryByText(/delete_outline/);
+    expect(deleteButton).not.toBeInTheDocument();
```

Comment:
Can also check that the edit and add buttons are not there.

Reply:
done